doc/add_kernel.rst

```diff
@@ -1,74 +1,52 @@
+   // file: MyGateImplementation.hpp
 
-    template <class PrecisionT>
     struct MyGateImplementation {
-        constexpr static implemented_gates = {GateOperations::PauliX};
+      public:
+        constexpr static std::array implemented_gates = {
+            GateOperation::PauliX
+        }; // List of implemented gates
         constexpr static kernel_id = KernelType::Mykernel; // Will be discussed below
+        constexpr static std::string_view = "MyGateImpl"; // Name of your kernel
 
+        template <class PrecisionT>
         static void applyPauliX(std::complex<PrecisionT>* data,
                                 size_t num_qubits,
                                 const std::vector<size_t>& wires,
                                 [[maybe_unused]] bool inverse) {
             /* Write your implementation */
             ...
         }
-
-        static void applyPauliY(std::complex<PrecisionT>* data,
-                                size_t num_qubits,
-                                const std::vector<size_t>& wires,
-                                [[maybe_unused]] bool inverse) {
-            PL_ABORT("MyGateImplementation::applyPauliY is not implemented");
-        }
-
-        /* All other gates */
-        ...
     };
 
-Note that all member functions must be defined to prevent compile errors (this requirement may be deprecated in the near future).
-
-Then you can add your gate implementation to Pennylane-Lightning by doing followings:
+Then you can add your gate implementation to Pennylane-Lightning. This can be done my modifying two files as:
 
 .. code-block:: cpp
 
     // file: simulator/KernelType.hpp
     namespace Pennylane {
-    enum class KernelType { PI, LM, MyKernel /* This is added */, Unknown };
-
-    namespace Constant {
-    constexpr std::array available_kernels = {
-        std::pair<KernelType, std::string_view>{KernelType::PI, "PI"},
-        std::pair<KernelType, std::string_view>{KernelType::LM, "LM"},
-        /* The following line is added */
-        std::pair<KernelType, std::string_view>{KernelType::MyKernel, "MyKernel"},
-    };
+    enum class KernelType { PI, LM, MyKernel /* This is added */, None };
 
     /* Rest of the file */
+
     } // namespace Pennylane
 
 and 
 
 .. code-block:: cpp
 
-    // file: simulator/SelectGateOps.hpp
+    // file: simulator/AvailableKernels.hpp
     namespace Pennylane {
-        ...
-        /* Some code */
-
-        template <class fp_t, KernelType kernel> class SelectGateOps {};
-
-        template <class fp_t>
-        class SelectGateOps<fp_t, KernelType::PI> : public GateOperationsPI<fp_t> {};
-        template <class fp_t>
-        class SelectGateOps<fp_t, KernelType::LM> : public GateOperationsLM<fp_t> {};
-
-        /* Add the following lines */
-        template <class fp_t>
-        class SelectGateOps<fp_t, KernelType::MyKernel> : public MyGateImplementation<fp_t> {};
+    using AvailableKernels = Util::TypeList<GateImplementationsLM,
```

Comment:
It may just be the Github formatting being off, but the indentation here looks a little short.

```suggestion
        using AvailableKernels = Util::TypeList<GateImplementationsLM,
```